internal/controller/notificationservice_controller_test.go

```diff
@@ -90,120 +95,199 @@ var _ = Describe("NotificationService Controller", func() {
+					err := fakeErrorNotifyNsr.SetupWithManager(k8sManager)
 					Expect(err).ToNot(HaveOccurred())
-					return metadata.HasAnnotationWithValue(createdPipelineRun, NotificationPipelineRunAnnotation, NotificationPipelineRunAnnotationValue)
-				}, timeout, interval).Should(BeTrue())
-				Expect(controllerutil.ContainsFinalizer(createdPipelineRun, NotificationPipelineRunFinalizer)).To(BeFalse())
-				// Check the Notify was called once
-				Expect(mn.Counter).To(Equal(1))
-			})
-		})
+					// The pipelinerun should be reconciled and the notification finalizer has been added successfully
+					Eventually(func() bool {
+						err := k8sClient.Get(ctx, pushPipelineRunLookupKey, createdPipelineRun)
+						Expect(err).ToNot(HaveOccurred())
+						return controllerutil.ContainsFinalizer(createdPipelineRun, NotificationPipelineRunFinalizer)
+					}, timeout, interval).Should(BeTrue())
+					Expect(controllerutil.ContainsFinalizer(createdPipelineRun, NotificationPipelineRunFinalizer)).To(BeTrue())
+					// Check the Notify was not called
+					Expect(mn.Counter).To(BeZero())
 
+					By("Updating status to completed successfully")
+					createdPipelineRun.Status = tektonv1.PipelineRunStatus{
+						PipelineRunStatusFields: tektonv1.PipelineRunStatusFields{
+							StartTime:      &metav1.Time{Time: time.Now()},
+							CompletionTime: &metav1.Time{Time: time.Now().Add(5 * time.Minute)},
+							Results: []tektonv1.PipelineRunResult{
+								{
+									Name:  "IMAGE_DIGEST",
+									Value: *tektonv1.NewStructuredValues("image_digest_value"),
+								},
+								{
+									Name:  "IMAGE_URL",
+									Value: *tektonv1.NewStructuredValues("image"),
+								},
+								{
+									Name:  "CHAINS-GIT_URL",
+									Value: *tektonv1.NewStructuredValues("git_url_value"),
+								},
+								{
+									Name:  "CHAINS-GIT_COMMIT",
+									Value: *tektonv1.NewStructuredValues("git_commit_value"),
+								},
+							},
+						},
+						Status: v1.Status{
+							Conditions: v1.Conditions{
+								apis.Condition{
+									Message: "Tasks Completed: 12 (Failed: 0, Cancelled 0), Skipped: 2",
+									Reason:  "Completed",
+									Status:  "True",
+									Type:    apis.ConditionSucceeded,
+								},
+							},
+						},
+					}
+					Expect(k8sClient.Status().Update(ctx, createdPipelineRun)).Should(Succeed())
+					// Wait for the Notify method to be called
+					Eventually(func() bool {
+						log.Println(fakeErrorNotify.Counter)
+						return fakeErrorNotify.Counter != 0
+					}, timeout, interval).Should(BeTrue())
+					Expect(controllerutil.ContainsFinalizer(createdPipelineRun, NotificationPipelineRunFinalizer)).To(BeTrue())
+					Expect(metadata.HasAnnotationWithValue(createdPipelineRun, NotificationPipelineRunAnnotation, NotificationPipelineRunAnnotationValue)).To(BeFalse())
+					// Since returning an error will reconcile the resource infintally, and we cannot track the number of Notify calls and
```

Comment:
```suggestion
					// Since returning an error will reconcile the resource infinitely, and we cannot track the number of Notify calls and
```
```suggestion
					// Since returning an error will reconcile the resource indefinitely, and we cannot track the number of Notify calls and
```